packages/apps/job-launcher/server/src/modules/job/job.service.spec.ts

```diff
@@ -305,6 +307,18 @@ describe('JobService', () => {
         save: jest.fn().mockResolvedValue(true),
       };
 
+      // Mock HMToken__factory.connect to return the mock HMToken
+      const mockTokenContract: any = {
+        transfer: jest.fn(),
+      };
+
+      jest.spyOn(HMToken__factory, 'connect').mockReturnValue(mockTokenContract);
+
+      jest.spyOn(mockTokenContract, 'transfer').mockResolvedValue({
+        chainId: 1,
+        hash: MOCK_TRANSACTION_HASH,
+      } as ContractTransaction);
+
```

Comment:
- add a check of the transfer being called
- add a test case where transfer fails